ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/text_embedding/TextEmbeddingModel.java

```diff
@@ -0,0 +1,239 @@
+                throw new IllegalArgumentException("unsupported engine");
+            }
+            AccessController.doPrivileged((PrivilegedExceptionAction<Void>) () -> {
+                ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+                try {
+                    System.setProperty("PYTORCH_PRECXX11", "true");
+                    System.setProperty("DJL_CACHE_DIR", DJL_CACHE_PATH.toAbsolutePath().toString());
+                    System.setProperty("java.library.path", DJL_CACHE_PATH.toAbsolutePath().toString());
+                    Thread.currentThread().setContextClassLoader(ai.djl.Model.class.getClassLoader());
+                    Path modelPath = getModelCachePath(modelId, modelName, version);
+                    File pathFile = new File(modelPath.toUri());
+                    if (pathFile.exists()) {
+                        FileUtils.deleteDirectory(pathFile);
+                    }
+                    ZipUtils.unzip(new FileInputStream(modelZipFile), modelPath);
+                    boolean findModelFile = false;
+                    for (File file : pathFile.listFiles()) {
+                        String name = file.getName();
+                        if (name.endsWith(PT) || name.endsWith(ONNX)) {
+                            if (findModelFile) {
+                                throw new IllegalArgumentException("found multiple models");
+                            }
+                            findModelFile = true;
+                            String suffix = name.substring(name.lastIndexOf("."));
+                            file.renameTo(new File(modelPath.resolve(modelName + suffix).toUri()));
+                        }
+                    }
+                    Map<String, Object> arguments = new HashMap<>();
+                    arguments.put("engine", engine);
+                    Criteria.Builder<Input, Output> criteriaBuilder = Criteria.builder()
+                            .setTypes(Input.class, Output.class)
+                            .optApplication(Application.UNDEFINED)
+                            .optArguments(arguments)
+                            .optModelPath(modelPath);
+                    TextEmbeddingModelConfig textEmbeddingModelConfig = (TextEmbeddingModelConfig) modelConfig;
+                    TextEmbeddingModelConfig.FrameworkType transformersType = textEmbeddingModelConfig.getFrameworkType();
+                    if (ONNX_ENGINE.equals(engine)) { //ONNX
+                        criteriaBuilder.optTranslator(new ONNXSentenceTransformerTextEmbeddingTranslator());
+                    } else { // pytorch
+                        if (transformersType == SENTENCE_TRANSFORMERS) {
+                            criteriaBuilder.optTranslator(new SentenceTransformerTextEmbeddingTranslator());
+                        } else {
+                            criteriaBuilder.optTranslatorFactory(new HuggingfaceTextEmbeddingTranslatorFactory());
```

Comment:
Is that possbile add a `serving.properties` in the zip file, so you don't have to write this ugly if else here.

Reply:
That's a good point. We can do that. We add transformer type in model for UI consideration, so user can review these model serving properties easily on model list page.

Put these serving properties in config file looks good for our future pre-built model or any model from model zoo. For current phase, we only support custom model. I think it's ok to leave it as is.